src/nodedef.cpp

```diff
@@ -682,6 +682,8 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 	switch (drawtype) {
 	default:
 	case NDT_NORMAL:
+		material_type = (alpha == 255) ?
```

Comment:
Corrected my last comment, mods that use 'use texture alpha' for normal nodes may be broken, but then they are using something that is buggy and unsupported anyway, and 0.5 is coming which gives us a good excuse.